src/initialize_dae.jl

```diff
@@ -564,7 +563,7 @@ function _initialize_dae!(integrator, prob::DAEProblem,
     if alg.nlsolve !== nothing
         nlsolve = alg.nlsolve
     else
-        nlsolve = NewtonRaphson(autodiff = isAD)
+        nlsolve = NewtonRaphson(autodiff = alg_autodiff(integrator.alg))
```

Comment:
We should keep it as type information now that we can use it effectively.

Reply:
Yes it changes the type level stuff to a bool as backwards compat support IIRC

Reply:
```
function alg_autodiff(alg)
    autodiff = _alg_autodiff(alg)
    if autodiff == Val(false)
        return AutoFiniteDiff()
    elseif autodiff == Val(true)
        return AutoForwardDiff()
    else
        return _unwrap_val(autodiff)
    end
end
```
This should constant propagate and give the correct AD type right?

Reply:
We shouldn't rely on constant propagation. It can fail. But let's pull this in and make this better in a follow up.